src/visitors/global_var_visitor.cpp

```diff
@@ -0,0 +1,35 @@
+/*************************************************************************
+ * Copyright (C) 2018-2019 Blue Brain Project
+ *
+ * This file is part of NMODL distributed under the terms of the GNU
+ * Lesser General Public License. See top-level LICENSE file for details.
+ *************************************************************************/
+
+#include <iostream>
+#include <memory>
+
+#include "ast/ast.hpp"
+#include "global_var_visitor.hpp"
+/**
+ * \file
+ * \brief AST Visitor to change nmodl::symtab::syminfo::NmodlType::global_var
+ * variables in nmodl::symtab to nmodl::symtab::syminfo::NmodlType::range_var
+ * so that they can be handled by Codegen as RANGE variables
+ */
+
+namespace nmodl {
+namespace visitor {
+
+using symtab::syminfo::NmodlType;
+
+void GlobalToRangeVisitor::visit_global_var(ast::GlobalVar* node) {
+    if (ast->get_symbol_table()->lookup(node->get_node_name())->get_write_count() > 0) {
+        ast->get_symbol_table()
+            ->lookup(node->get_node_name())
+            ->remove_property(NmodlType::global_var);
+        ast->get_symbol_table()->lookup(node->get_node_name())->add_property(NmodlType::range_var);
```

Comment:
I think you two days ago you asked half question : `can we change node type...`. Is it related to here? i.e. Can we change the type of node in the AST?

Answer is yes. In above case you have merely changed the property of node in the AST (underlying node is still GLOBAL). 

What we have to do is, we have to remove variables from GLOBAL node and add them to RANGE node. (if RANGE node doesn't exist we have to create one).